react-forms/my-app/src/ControlledForm.tsx

```diff
@@ -0,0 +1,35 @@
+import { FormEvent, useState } from 'react';
+
+export function RegistrationFormControlled() {
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+
+  function handleSubmit(event: FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+    console.log('ControlledFormObject', { username, password });
+  }
+
+  return (
+    <form id="controlledForm" onSubmit={handleSubmit}>
+      <label>
+        Username:{' '}
+        <input
+          id="username"
+          type="text"
+          name="username"
```

Comment:
In a controlled form, the `name` attribute on these inputs isn't necessary. It doesn't hurt to have them, but leaving them off saves some typing.